src/bun.js/bindings/JSBuffer.cpp

```diff
@@ -2400,8 +2414,15 @@ JSC_DEFINE_HOST_FUNCTION(constructJSBuffer, (JSC::JSGlobalObject * lexicalGlobal
 
             RELEASE_AND_RETURN(throwScope, JSC::JSValue::encode(uint8Array));
         }
-        default: {
+        case ArrayType:
+        case DerivedArrayType:
             break;
+        default: {
```

Comment:
What happens if it's an iterable? What happens if its an object which implements arraySpecies?